src/wcc/gen_wasm.c

```diff
@@ -2138,6 +2138,60 @@ static void gen_alloca(Expr *expr, enum BuiltinFunctionPhase phase) {
   gen_expr(result, true);
 }
 
+static void gen_builtin_clz(Expr *expr, enum BuiltinFunctionPhase phase) {
+  assert(expr->kind == EX_FUNCALL);
+  Vector *args = expr->funcall.args;
+  assert(args->len == 1);
+  Expr *value = args->data[0];
+
+  switch (phase) {
+  case BFP_TRAVERSE:
+    if (!is_fixnum(value->type->kind) || type_size(value->type) > I32_SIZE)
```

Comment:
Argument for builtin function is type checked.
So you don't need traverse phase.